server/src/main/java/org/opensearch/search/SearchService.java

```diff
@@ -1114,6 +1138,56 @@ private DefaultSearchContext createSearchContext(ReaderContext reader, ShardSear
         return searchContext;
     }
 
+    private void writeTaskResourceUsage(SearchShardTask task) {
+        try {
+            // Get resource usages from when the task started
+            ThreadResourceInfo threadResourceInfo = task.getActiveThreadResourceInfo(
+                Thread.currentThread().getId(),
+                ResourceStatsType.WORKER_STATS
+            );
```

Comment:
This logic should also be in the TaskResourceTrackingService?

Reply:
hmm, let me refactor this part.